syntax = "proto3";

package TW.Aion.Proto;
option java_package = "wallet.core.jni.proto";

// Input data necessary to create a signed transaction.
message SigningInput {
    // Nonce (256-bit number)
    bytes nonce = 1;

    // Gas price (256-bit number)
    bytes gas_price = 2;

    // Gas limit (256-bit number)
    bytes gas_limit = 3;

    // Recipient's address.
    string to_address = 4;

    // Amount to send in wei (256-bit number)
    bytes amount = 5;

    // Optional payload
    bytes payload = 6;

    // Private key.
    bytes private_key = 7;
}

// Transaction signing output.
message SigningOutput {
    // Signed and encoded transaction bytes.
    bytes encoded = 1;

    // Signature.
    bytes signature = 2;
}
